src/reconcile/rec_write.c

```diff
@@ -2367,14 +2375,32 @@ __rec_build_delta_leaf(WT_SESSION_IMPL *session, WT_PAGE_HEADER *full_image, WTI
     WT_RET(__wti_rec_build_delta_init(session, r));
 
     for (i = 0, supd = multi->supd; i < multi->supd_entries; ++i, ++supd) {
-        if (supd->onpage_upd == NULL)
-            continue;
-
-        if (supd->onpage_tombstone != NULL && F_ISSET(supd->onpage_tombstone, WT_UPDATE_DURABLE))
+        if (supd->onpage_upd == NULL && supd->onpage_tombstone == NULL)
             continue;
 
-        if (supd->onpage_tombstone == NULL && F_ISSET(supd->onpage_upd, WT_UPDATE_DURABLE))
-            continue;
+        if (supd->onpage_upd == NULL) {
```

Comment:
Here also a comment would help the reader.

Reply:
added